shared/lib/accounts/accounts.ts

```diff
@@ -35,46 +50,192 @@ export function getUniqueAccountName(
   return candidateName;
 }
 
-///: BEGIN:ONLY_INCLUDE_IF(solana)
-export async function addDiscoveredSolanaAccounts(
-  controllerMessenger: KeyringInternalSnapClientMessenger,
-  entropySource: string,
-  snapKeyring: SnapKeyring,
+export type SnapAccountNameOptions = {
+  chainId?: CaipChainId;
+};
+
+/**
+ * Get the next available Snap account name for a supported non-EVM Snap.
+ *
+ * @param getNextAvailableAccountName - Callback to get the next available account name.
+ * @param snapId - Snap ID.
+ * @param options - Options for this account name.
+ * @param options.chainId - Chain ID for this account if available.
+ * @returns
+ */
+export async function getNextAvailableSnapAccountName(
+  getNextAvailableAccountName: () => Promise<string>,
+  snapId: SUPPORTED_WALLET_SNAP_ID,
+  { chainId }: SnapAccountNameOptions = {},
 ) {
-  const snapId = SOLANA_WALLET_SNAP_ID;
-  const scopes = [SolScope.Mainnet, SolScope.Testnet, SolScope.Devnet];
-  const client = new KeyringInternalSnapClient({
-    messenger: controllerMessenger,
-    snapId,
-  });
-
-  for (let index = 0; ; index++) {
-    const discovered = await client.discoverAccounts(
-      scopes,
-      entropySource,
-      index,
-    );
+  const defaultSnapAccountName = await getNextAvailableAccountName();
+
+  // FIXME: This is a temporary workaround to suggest a different account name for a first party snap.
+  const accountNumber = defaultSnapAccountName.trim().split(' ').pop();
```

Comment:
In this case, the snap names will increase regardless of the snap type since we're getting the next available name based on the keyring type, for example, `Bitcoin 1`, `Solana 2`, `Bitcoin 3`, etc?

Reply:
This is correct yes. For now, we decide to go for a temporary solution until we move/refactor the naming logic elsewhere.

That's not new BTW, it's been like this since:
- https://github.com/MetaMask/metamask-extension/pull/31542